app/assets/javascripts/modules/inline-image-modal.js

```diff
@@ -50,6 +75,17 @@ InlineImageModal.prototype.performAction = function (item) {
       var editor = this.$module.closest('[data-module="markdown-editor"]')
       this.$modal.close()
       editor.selectionReplace(item.dataset.modalData)
+    },
+    'upload': function () {
+      this.postModalForm(item.dataset.modalActionForm)
```

Comment:
Are we vulnerable at this point to dual submissions? I assume that nothing changes in the UI once they submit the form until we receive a response which could be a while for an upload.

Could well be that we have the same problem with the non modal form of this too but feedback seems to be clearer in a browser when you're changing a whole page.

Reply:
So we are actually changing the modal to the loading screen - this happens before any action is invoked, as otherwise it kind of defeats the point of the loading screen ;-). I think there's a separate question, which you also asked, about the behaviour of short-lived loading screens.

I guess they can potentially click the button twice, really quickly, although I can't do this myself - in a very slow browser, perhaps.

Reply:
Oh ok - I think I might just not be understanding how this works. Is this correct?

- When I click the button to submit the form it prevents default on that event
- It focuses the modal again
- It hides the form and shows the loading view
- it pulls the data from the hidden view and submits it
- that form is only cleared once showDynamic section is called again?

Reply:
Yes, that's right - is there still something you're concerned about here?

Reply:
Nope - just taking my old brain a while to understand